nope, this has no effect!
diff --git a/modules/Makefile.am b/modules/Makefile.am
index 5e2151a051..53b90a2713 100644
--- a/modules/Makefile.am
+++ b/modules/Makefile.am
@@ -3,7 +3,6 @@ check_LTLIBRARIES =
 pkglib_LTLIBRARIES =
 noinst_HEADERS =
 check_PROGRAMS =
-pkglibexec_PROGRAMS =
 EXTRA_DIST =
 
 EXTRA_SUBDIRS = \
diff --git a/modules/gui/qt/Makefile.am b/modules/gui/qt/Makefile.am
index a48677be92..cb77b14d96 100644
--- a/modules/gui/qt/Makefile.am
+++ b/modules/gui/qt/Makefile.am
@@ -448,14 +448,7 @@ nodist_libqt_plugin_la_SOURCES += gui/qt/resources.cpp
 gui/qt/resources.cpp: gui/qt/vlc.qrc $(libqt_plugin_la_RES)
 	$(AM_V_GEN)$(RCC) -name vlc -o $@ $<
 
-vlc_qt_check_SOURCES = gui/qt/vlc-qt-check.cpp
-vlc_qt_check_CXXFLAGS = $(AM_CXXFLAGS) $(QT_CFLAGS) -fPIC
-vlc_qt_check_LDADD = $(QT_LIBS)
-
 if ENABLE_QT
 gui_LTLIBRARIES += libqt_plugin.la
 BUILT_SOURCES += $(nodist_libqt_plugin_la_SOURCES)
-if !HAVE_WIN32
-pkglibexec_PROGRAMS += vlc-qt-check
-endif
 endif
* Unmerged path modules/gui/qt/vlc-qt-check.cpp
