diff --git a/sys-utils/dmesg.c b/sys-utils/dmesg.c
index 1eb7cde3c..ccda80b48 100644
--- a/sys-utils/dmesg.c
+++ b/sys-utils/dmesg.c
@@ -661,9 +661,9 @@ static void safe_fwrite(struct dmesg_control *ctl, const char *buf, size_t size,
 		}
 
 doprint:
-		if (hex)
+		/*if (hex)
 			rc = fwrite_hex(p, len, out);
-		else if (*p == '\n' && *(p + 1) && indent) {
+		else*/ if (*p == '\n' && *(p + 1) && indent) {
 		        rc = fwrite(p, 1, len, out) != len;
 			if (fprintf(out, "%*s", indent, "") != indent)
 				rc |= 1;
#diff --git a/sys-utils/dmesg.c b/sys-utils/dmesg.c
#index 9f2e5b804..e0c432833 100644
#--- a/sys-utils/dmesg.c
#+++ b/sys-utils/dmesg.c
#@@ -651,9 +651,9 @@ static void safe_fwrite(const char *buf, size_t size, int indent, FILE *out)
# 				hex = 1;
# 		}
# 
#-		if (hex)
#+		/*if (hex)
# 			rc = fwrite_hex(p, len, out);
#-		else if (*p == '\n' && *(p + 1) && indent) {
#+		else */if (*p == '\n' && *(p + 1) && indent) {
# 		        rc = fwrite(p, 1, len, out) != len;
# 			if (fprintf(out, "%*s", indent, "") != indent)
# 				rc |= 1;
