diff --git a/src/cargo/util/workspace.rs b/src/cargo/util/workspace.rs
index 73ee04499..88d32ee9f 100644
--- a/src/cargo/util/workspace.rs
+++ b/src/cargo/util/workspace.rs
@@ -39,7 +39,7 @@ fn print_available(
     writeln!(output, "\"{}\" takes one argument.", option_name)?;
 
     if targets.is_empty() {
-        writeln!(output, "No {} available.", plural_name)?;
+        writeln!(output, "No {} available. So you can't pass any arg to `--example` at this time!", plural_name)?;
     } else {
         writeln!(output, "Available {}:", plural_name)?;
         for target in targets {
