https://github.com/gwsw/less/issues/57

Index: less-551/search.c
===================================================================
--- less-551.orig/search.c
+++ less-551/search.c
@@ -1021,6 +1025,8 @@ hilite_line(linepos, line, line_len, chp
 			searchp++;
 		else /* end of line */
 			break;
+		if (ABORT_SIGS())
+			break;
 	} while (match_pattern(info_compiled(&search_info), search_info.text,
 			searchp, line_end - searchp, &sp, &ep, 1, search_info.search_type));
 }
