diff --git a/src/pulsecore/log.c b/src/pulsecore/log.c
index 2ea88620b..418fc243a 100644
--- a/src/pulsecore/log.c
+++ b/src/pulsecore/log.c
@@ -90,7 +90,7 @@ static bool target_override_set = false;
 static pa_log_level_t maximum_level = PA_LOG_ERROR, maximum_level_override = PA_LOG_ERROR;
 static unsigned show_backtrace = 0, show_backtrace_override = 0, skip_backtrace = 0;
 static pa_log_flags_t flags = 0, flags_override = 0;
-static bool no_rate_limit = false;
+static bool no_rate_limit = true; //temp
 static int log_fd = -1;
 static int write_type = 0;
 
