diff --git a/Makefile.def b/Makefile.def
index 28bf61d771..4058fae920 100644
--- a/Makefile.def
+++ b/Makefile.def
@@ -112,7 +112,7 @@ host_modules= { module= texinfo; no_install= true; };
 host_modules= { module= zlib; no_install=true; no_check=true;
 		bootstrap=true;
 	        extra_configure_flags='@extra_host_zlib_configure_flags@';};
-host_modules= { module= gnulib; };
+host_modules= { module= gnulib; no_install= true; no_check=true; };
 host_modules= { module= gdb; };
 host_modules= { module= expect; };
 host_modules= { module= guile; };
