diff --git a/bashline.c b/bashline.c
index cd36b0be..78820b00 100644
--- a/bashline.c
+++ b/bashline.c
@@ -4688,7 +4688,8 @@ bash_event_hook ()
   bashline_reset_event_hook ();
 
   /* posix mode SIGINT during read -e. We only get here if SIGINT is trapped. */
-  if (posixly_correct && this_shell_builtin == read_builtin && sig == 2)
+  if (/*posixly_correct && */ // not only when POSIX, thanks to Chet Ramey for the info about this here: https://lists.gnu.org/archive/html/bug-bash/2020-04/msg00110.html
+        this_shell_builtin == read_builtin && sig == 2)
     {
       last_command_exit_value = 128|SIGINT;
       throw_to_top_level ();
