diff --git a/panel-plugin/time-out.c b/panel-plugin/time-out.c
index d537d87..8aad22b 100644
--- a/panel-plugin/time-out.c
+++ b/panel-plugin/time-out.c
@@ -154,6 +154,15 @@ static void           time_out_lock_countdown_finish              (TimeOutCountd
 XFCE_PANEL_PLUGIN_REGISTER (time_out_construct);
 
 
+static int
+hide_or_display_panel_icon(TimeOutPlugin   *time_out) {
+  if (time_out->display_icon)
+    gtk_widget_show (time_out->panel_icon);
+  else if (time_out->display_time)
+    gtk_widget_hide (time_out->panel_icon);
+  else return 1;
+  return 0;
+}
 
 static TimeOutPlugin*
 time_out_new (XfcePanelPlugin *plugin)
@@ -209,7 +218,8 @@ time_out_new (XfcePanelPlugin *plugin)
   gtk_image_set_pixel_size (GTK_IMAGE (time_out->panel_icon), xfce_panel_plugin_get_size (time_out->plugin) - 8);
 #endif
   gtk_box_pack_start (GTK_BOX (time_out->hvbox), time_out->panel_icon, TRUE, TRUE, 0);
-  gtk_widget_show (time_out->panel_icon);
+  //gtk_widget_show (time_out->panel_icon);
+  //hide_or_display_panel_icon (time_out);//not here, settings aren't loaded yet!
 
   /* Create label for displaying the remaining time until the next break */
   time_out->time_label = gtk_label_new (_("Inactive"));
@@ -833,12 +843,14 @@ time_out_display_icon_toggled (GtkToggleButton *toggle_button,
   time_out->display_icon = gtk_toggle_button_get_active (toggle_button);
 
   /* Hide or display the panel icon */
-  if (time_out->display_icon)
-    gtk_widget_show (time_out->panel_icon);
-  else if (time_out->display_time)
-    gtk_widget_hide (time_out->panel_icon);
-  else
+//  if (time_out->display_icon)
+//    gtk_widget_show (time_out->panel_icon);
+//  else if (time_out->display_time)
+//    gtk_widget_hide (time_out->panel_icon);
+//  else
+  if (hide_or_display_panel_icon(time_out) == 1) {
     gtk_toggle_button_set_active(toggle_button, TRUE);
+  }
 }
 
 
@@ -906,6 +918,8 @@ time_out_load_settings (TimeOutPlugin *time_out)
   time_out->display_icon = display_icon;
   time_out->allow_postpone = allow_postpone;
   time_out->auto_resume = auto_resume;
+
+  hide_or_display_panel_icon (time_out);
 }
 
 
