segfaults X
diff --git a/src/drmmode_display.c b/src/drmmode_display.c
index 8445ef2a..8d530ed4 100644
--- a/src/drmmode_display.c
+++ b/src/drmmode_display.c
@@ -967,8 +967,8 @@ done:
 		if (drmmode_crtc->scanout[scanout_id].pixmap &&
 		    fb != radeon_pixmap_get_fb(drmmode_crtc->
 					       scanout[scanout_id].pixmap)) {
-			radeon_drm_abort_entry(drmmode_crtc->scanout_update_pending);
-			drmmode_crtc->scanout_update_pending = 0;
+			drmmode_crtc_wait_pending_event(drmmode_crtc, pRADEONEnt->fd,
+							drmmode_crtc->scanout_update_pending);
 			drmmode_crtc_scanout_free(drmmode_crtc);
 		} else if (!drmmode_crtc->tear_free) {
 			drmmode_crtc_scanout_destroy(drmmode,
@@ -3359,12 +3359,14 @@ Bool radeon_do_pageflip(ScrnInfoPtr scrn, ClientPtr client,
 						 extents);
 			radeon_cs_flush_indirect(crtc->scrn);
 
-			if (drmmode_crtc->scanout_update_pending) {
-				radeon_drm_wait_pending_flip(crtc);
-				handle_deferred = TRUE;
-				radeon_drm_abort_entry(drmmode_crtc->scanout_update_pending);
-				drmmode_crtc->scanout_update_pending = 0;
-			}
+//			if (drmmode_crtc->scanout_update_pending) {
+//				radeon_drm_wait_pending_flip(crtc);
+//				handle_deferred = TRUE;
+//				radeon_drm_abort_entry(drmmode_crtc->scanout_update_pending);
+//				drmmode_crtc->scanout_update_pending = 0;
+//			}
+      drmmode_crtc_wait_pending_event(drmmode_crtc, pRADEONEnt->fd,
+             drmmode_crtc->scanout_update_pending);
 		} else {
 			drmmode_fb_reference(pRADEONEnt->fd, &flipdata->fb[i], fb);
 		}
