Index: /var/tmp/portage/www-client/firefox-97.0.2/work/firefox-97.0.2/browser/modules/AboutNewTab.jsm
===================================================================
--- firefox-97.0.2/browser/modules/AboutNewTab.jsm
+++ firefox-97.0.2/browser/modules/AboutNewTab.jsm
@@ -14,7 +14,7 @@ const { XPCOMUtils } = ChromeUtils.impor
 const { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
 
 XPCOMUtils.defineLazyModuleGetters(this, {
-  ActivityStream: "resource://activity-stream/lib/ActivityStream.jsm",
+  //ActivityStream: "resource://activity-stream/lib/ActivityStream.jsm",
   ObjectUtils: "resource://gre/modules/ObjectUtils.jsm",
   RemotePages:
     "resource://gre/modules/remotepagemanager/RemotePageManagerParent.jsm",
@@ -172,6 +172,7 @@ const AboutNewTab = {
    * onBrowserReady - Continues the initialization of Activity Stream after browser is ready.
    */
   onBrowserReady() {
+    return;
     if (this.activityStream && this.activityStream.initialized) {
       return;
     }

#diff -r e704e3565db9 browser/modules/AboutNewTab.jsm
#--- a/browser/modules/AboutNewTab.jsm	Thu Jul 11 00:49:58 2019 +0300
#+++ b/browser/modules/AboutNewTab.jsm	Thu Jul 11 03:42:03 2019 +0200
#@@ -12,7 +12,7 @@
# const { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
# 
# XPCOMUtils.defineLazyModuleGetters(this, {
#-  ActivityStream: "resource://activity-stream/lib/ActivityStream.jsm",
#+  //ActivityStream: "resource://activity-stream/lib/ActivityStream.jsm",
#   RemotePages:
#     "resource://gre/modules/remotepagemanager/RemotePageManagerParent.jsm",
# });
#@@ -38,7 +38,7 @@
#   willNotifyUser: false,
# 
#   _activityStreamEnabled: false,
#-  activityStream: null,
#+  //activityStream: null,
#   activityStreamDebug: false,
# 
#   _newTabURL: ABOUT_URL,
#@@ -63,16 +63,16 @@
#    * onBrowserReady - Continues the initialization of Activity Stream after browser is ready.
#    */
#   onBrowserReady() {
#-    if (this.activityStream && this.activityStream.initialized) {
#-      return;
#-    }
#-
#-    this.activityStream = new ActivityStream();
#-    try {
#-      this.activityStream.init();
#-    } catch (e) {
#-      Cu.reportError(e);
#-    }
#+//    if (this.activityStream && this.activityStream.initialized) {
#+//      return;
#+//    }
#+//
#+//    this.activityStream = new ActivityStream();
#+//    try {
#+//      this.activityStream.init();
#+//    } catch (e) {
#+//      Cu.reportError(e);
#+//    }
#   },
# 
#   /**
#@@ -80,10 +80,10 @@
#    *        if it exists.
#    */
#   uninit() {
#-    if (this.activityStream) {
#-      this.activityStream.uninit();
#-      this.activityStream = null;
#-    }
#+//    if (this.activityStream) {
#+//      this.activityStream.uninit();
#+//      this.activityStream = null;
#+//    }
# 
#     if (this.pageListener) {
#       this.pageListener.destroy();
