diff -r dc5251d30a38 old-configure.in
--- a/old-configure.in	Tue Apr 14 00:32:50 2020 +0000
+++ b/old-configure.in	Tue Apr 14 12:40:49 2020 +0200
@@ -2911,10 +2911,11 @@
 AC_SUBST(MOZ_PKG_SPECIAL)
 AC_SUBST(MOZ_SIMPLE_PACKAGE_NAME)
 
-if test "$MOZILLA_OFFICIAL"; then
+#if test "$MOZILLA_OFFICIAL"; then
     # Build revisions should always be present in official builds
     MOZ_INCLUDE_SOURCE_INFO=1
-fi
+    #this will define&show MOZ_SOURCE_URL in about:buildconfig
+#fi
 
 # External builds (specifically Ubuntu) may drop the hg repo information, so we allow to
 # explicitly set the repository and changeset information in.
