Index: /var/tmp/portage/sys-apps/portage-3.0.35/work/portage-3.0.35/bin/ebuild.sh
===================================================================
--- portage-3.0.35/bin/ebuild.sh
+++ portage-3.0.35/bin/ebuild.sh
@@ -255,6 +255,9 @@ inherit() {
 				break
 			fi
 		done
+		if [[ "$location" =~ ^/eclasses_override/ ]]; then
+			eqawarn "Using(inherit) modified eclass '$location'"
+		fi
 		debug-print "inherit: ${1} -> ${location}"
 		[[ -z ${location} ]] && die "${1}.eclass could not be found by inherit()"
 
