diff --git a/arch/x86/kernel/cpu/microcode/core.c b/arch/x86/kernel/cpu/microcode/core.c
index 9f0fe5bb450d..c1861c191568 100644
--- a/arch/x86/kernel/cpu/microcode/core.c
+++ b/arch/x86/kernel/cpu/microcode/core.c
@@ -620,8 +620,10 @@ static ssize_t reload_store(struct devic
 		return size;
 
 	tmp_ret = microcode_ops->request_microcode_fw(bsp, &microcode_pdev->dev, true);
-	if (tmp_ret != UCODE_NEW)
+	if (tmp_ret != UCODE_NEW) {
+		pr_info("Not reloading previously-loaded already-in-effect microcode!\n");//not hit if no early loading happened before, even after sudo sh -c "echo 1  > /sys/devices/system/cpu/microcode/reload"
 		return size;
+  }
 
 	get_online_cpus();
 
#diff --git a/arch/x86/kernel/cpu/microcode/amd.c b/arch/x86/kernel/cpu/microcode/amd.c
#index a998e1a7d46f..48179928ff38 100644
#--- a/arch/x86/kernel/cpu/microcode/amd.c
#+++ b/arch/x86/kernel/cpu/microcode/amd.c
#@@ -699,8 +699,10 @@ load_microcode_amd(bool save, u8 family,
# 	if (!p) {
# 		return ret;
# 	} else {
#-		if (boot_cpu_data.microcode == p->patch_id)
#+		if (boot_cpu_data.microcode == p->patch_id) {
#+			pr_debug("Already in-memory microcode 0x%x ret=%d\n", boot_cpu_data.microcode, ret);//not hit when no early loading happened at all, in dmesg even after sudo sh -c "echo 1  > /sys/devices/system/cpu/microcode/reload"
# 			return ret;
#+    }
# 
# 		ret = UCODE_NEW;
# 	}
diff --git a/arch/x86/kernel/cpu/microcode/amd.c b/arch/x86/kernel/cpu/microcode/amd.c
index e1f3ba19ba54..878d1a8d2642 100644
--- a/arch/x86/kernel/cpu/microcode/amd.c
+++ b/arch/x86/kernel/cpu/microcode/amd.c
@@ -855,8 +855,10 @@ load_microcode_amd(bool save, u8 family, const u8 *data, size_t size)
 	if (!p) {
 		return ret;
 	} else {
-		if (boot_cpu_data.microcode >= p->patch_id)
+		if (boot_cpu_data.microcode >= p->patch_id) {
+      pr_debug("Already in-memory microcode 0x%x ret=%d\n", boot_cpu_data.microcode, ret);//not hit when no early loading happened at all, in dmesg even after sudo sh -c "echo 1  > /sys/devices/system/cpu/microcode/reload"
 			return ret;
+    }
 
 		ret = UCODE_NEW;
 	}
