diff --git a/tests/src/filemanager/Makefile.am b/tests/src/filemanager/Makefile.am
index 19a9ba955..eb328bd4d 100644
--- a/tests/src/filemanager/Makefile.am
+++ b/tests/src/filemanager/Makefile.am
@@ -31,7 +31,8 @@ TESTS = \
 	examine_cd \
 	exec_get_export_variables_ext \
 	filegui_is_wildcarded \
-	get_random_hint
+	get_random_hint \
+	compare_directories
 
 check_PROGRAMS = $(TESTS)
 
@@ -49,3 +50,7 @@ get_random_hint_SOURCES = \
 
 filegui_is_wildcarded_SOURCES = \
 	filegui_is_wildcarded.c
+
+compare_directories_SOURCES = \
+	compare_directories.c
+
