https://bugzilla.xfce.org/show_bug.cgi?id=16426

diff --git a/libxfce4panel/xfce-panel-plugin.c b/libxfce4panel/xfce-panel-plugin.c
index 1e11b6ed..9c2396d4 100644
--- a/libxfce4panel/xfce-panel-plugin.c
+++ b/libxfce4panel/xfce-panel-plugin.c
@@ -371,7 +371,7 @@ xfce_panel_plugin_class_init (XfcePanelPluginClass *klass)
                   G_STRUCT_OFFSET (XfcePanelPluginClass, remote_event),
                   NULL, NULL,
                   _libxfce4panel_marshal_BOOLEAN__STRING_BOXED,
-                  G_TYPE_BOOLEAN, 2, G_TYPE_STRING, G_TYPE_VALUE);
+                  G_TYPE_BOOLEAN, 3, G_TYPE_STRING, G_TYPE_VALUE, G_TYPE_UINT);
 
   /**
    * XfcePanelPlugin::removed
@@ -1728,7 +1728,7 @@ xfce_panel_plugin_remote_event (XfcePanelPluginProvider *provider,
   panel_return_val_if_fail (value == NULL || G_IS_VALUE (value), TRUE);
 
   g_signal_emit (G_OBJECT (provider), plugin_signals[REMOTE_EVENT], 0,
-                 name, value, &stop_emission);
+                 name, value, handle, &stop_emission);
 
   return stop_emission;
 }
diff --git a/libxfce4panel/xfce-panel-plugin.h b/libxfce4panel/xfce-panel-plugin.h
index 6cfe47cf..f97504cc 100644
--- a/libxfce4panel/xfce-panel-plugin.h
+++ b/libxfce4panel/xfce-panel-plugin.h
@@ -124,7 +124,8 @@ struct _XfcePanelPluginClass
   void     (*removed)                 (XfcePanelPlugin    *plugin);
   gboolean (*remote_event)            (XfcePanelPlugin    *plugin,
                                        const gchar        *name,
-                                       const GValue       *value);
+                                       const GValue       *value,
+                                       guint              *handle);
 
   /* new in 4.10 */
   void     (*mode_changed)            (XfcePanelPlugin    *plugin,

diff --git a/plugins/applicationsmenu/applicationsmenu.c b/plugins/applicationsmenu/applicationsmenu.c
index f099e1af..4da753ca 100644
--- a/plugins/applicationsmenu/applicationsmenu.c
+++ b/plugins/applicationsmenu/applicationsmenu.c
@@ -108,7 +108,8 @@ static void      applications_menu_plugin_mode_changed         (XfcePanelPlugin
 static void      applications_menu_plugin_configure_plugin     (XfcePanelPlugin        *panel_plugin);
 static gboolean  applications_menu_plugin_remote_event         (XfcePanelPlugin        *panel_plugin,
                                                                 const gchar            *name,
-                                                                const GValue           *value);
+                                                                const GValue           *value,
+                                                                guint              *handle);
 static gboolean  applications_menu_plugin_menu                 (GtkWidget              *button,
                                                                 GdkEventButton         *event,
                                                                 ApplicationsMenuPlugin *plugin);
@@ -736,7 +737,8 @@ applications_menu_plugin_configure_plugin (XfcePanelPlugin *panel_plugin)
 static gboolean
 applications_menu_plugin_remote_event (XfcePanelPlugin *panel_plugin,
                                        const gchar     *name,
-                                       const GValue    *value)
+                                       const GValue    *value,
+                                       guint              *handle)
 {
   ApplicationsMenuPlugin *plugin = XFCE_APPLICATIONS_MENU_PLUGIN (panel_plugin);
 
diff --git a/plugins/directorymenu/directorymenu.c b/plugins/directorymenu/directorymenu.c
index 4bddec14..1d10128f 100644
--- a/plugins/directorymenu/directorymenu.c
+++ b/plugins/directorymenu/directorymenu.c
@@ -100,7 +100,8 @@ static gboolean  directory_menu_plugin_size_changed         (XfcePanelPlugin
 static void      directory_menu_plugin_configure_plugin     (XfcePanelPlugin     *panel_plugin);
 static gboolean  directory_menu_plugin_remote_event         (XfcePanelPlugin     *panel_plugin,
                                                              const gchar         *name,
-                                                             const GValue        *value);
+                                                             const GValue        *value,
+                                                             guint              *handle);
 static void      directory_menu_plugin_menu                 (GtkWidget           *button,
                                                              DirectoryMenuPlugin *plugin);
 
@@ -572,7 +573,8 @@ directory_menu_plugin_configure_plugin (XfcePanelPlugin *panel_plugin)
 static gboolean
 directory_menu_plugin_remote_event (XfcePanelPlugin *panel_plugin,
                                     const gchar     *name,
-                                    const GValue    *value)
+                                    const GValue    *value,
+                                    guint              *handle)
 {
   DirectoryMenuPlugin *plugin = XFCE_DIRECTORY_MENU_PLUGIN (panel_plugin);
 
diff --git a/plugins/launcher/launcher.c b/plugins/launcher/launcher.c
index d21306bb..689cc1a6 100644
--- a/plugins/launcher/launcher.c
+++ b/plugins/launcher/launcher.c
@@ -61,7 +61,8 @@ static void               launcher_plugin_free_data                     (XfcePan
 static void               launcher_plugin_removed                       (XfcePanelPlugin      *panel_plugin);
 static gboolean           launcher_plugin_remote_event                  (XfcePanelPlugin      *panel_plugin,
                                                                          const gchar          *name,
-                                                                         const GValue         *value);
+                                                                         const GValue         *value,
+                                                                         guint              *handle);
 static gboolean           launcher_plugin_save_delayed_timeout          (gpointer              user_data);
 static void               launcher_plugin_save_delayed                  (LauncherPlugin       *plugin);
 static void               launcher_plugin_mode_changed                  (XfcePanelPlugin      *panel_plugin,
@@ -1208,7 +1209,8 @@ launcher_plugin_removed (XfcePanelPlugin *panel_plugin)
 static gboolean
 launcher_plugin_remote_event (XfcePanelPlugin *panel_plugin,
                               const gchar     *name,
-                              const GValue    *value)
+                              const GValue    *value,
+                              guint              *handle)
 {
   LauncherPlugin *plugin = XFCE_LAUNCHER_PLUGIN (panel_plugin);
 
diff --git a/plugins/windowmenu/windowmenu.c b/plugins/windowmenu/windowmenu.c
index ab0a40e9..4e7b7bc4 100644
--- a/plugins/windowmenu/windowmenu.c
+++ b/plugins/windowmenu/windowmenu.c
@@ -110,7 +110,8 @@ static gboolean  window_menu_plugin_size_changed            (XfcePanelPlugin
 static void      window_menu_plugin_configure_plugin        (XfcePanelPlugin    *panel_plugin);
 static gboolean  window_menu_plugin_remote_event            (XfcePanelPlugin    *panel_plugin,
                                                              const gchar        *name,
-                                                             const GValue       *value);
+                                                             const GValue       *value,
+                                                             guint              *handle);
 static void      window_menu_plugin_active_window_changed   (WnckScreen         *screen,
                                                              WnckWindow         *previous_window,
                                                              WindowMenuPlugin   *plugin);
@@ -583,7 +584,8 @@ window_menu_plugin_configure_plugin (XfcePanelPlugin *panel_plugin)
 static gboolean
 window_menu_plugin_remote_event (XfcePanelPlugin *panel_plugin,
                                  const gchar     *name,
-                                 const GValue    *value)
+                                 const GValue    *value,
+                                 guint              *handle)
 {
   WindowMenuPlugin *plugin = XFCE_WINDOW_MENU_PLUGIN (panel_plugin);
 
