--- a/google_apis/gaia/gaia_urls.cc	2019-06-06 20:51:29.000000000 +0200
+++ b/google_apis/gaia/gaia_urls.cc	2019-06-15 19:56:54.928940763 +0200
@@ -13,11 +13,11 @@
 namespace {
 
 // Gaia service constants
-const char kDefaultGoogleUrl[] = "http://google.com";
-const char kDefaultGaiaUrl[] = "https://accounts.google.com";
-const char kDefaultGoogleApisBaseUrl[] = "https://www.googleapis.com";
+const char kDefaultGoogleUrl[] = "http://google.com.BLOCKEDfromsourcecodeDOMAIN";
+const char kDefaultGaiaUrl[] = "https://accounts.google.com.BLOCKEDfromsourcecodeDOMAIN";
+const char kDefaultGoogleApisBaseUrl[] = "https://www.googleapis.com.BLOCKEDfromsourcecodeDOMAIN";
 const char kDefaultOAuthAccountManagerBaseUrl[] =
-    "https://oauthaccountmanager.googleapis.com";
+    "https://oauthaccountmanager.googleapis.com.BLOCKEDfromsourcecodeDOMAIN";
 
 // API calls from accounts.google.com
 const char kClientLoginUrlSuffix[] = "ClientLogin";
