bug: https://bugzilla.kernel.org/show_bug.cgi?id=208221
    commit 7778d8417b74aded842eeb372961cfc460417fa0 upstream.

diff --git a/arch/x86/include/asm/vdso/gettimeofday.h b/arch/x86/include/asm/vdso/gettimeofday.h
index fb81fea99093..9a6dc9b4ec99 100644
--- a/arch/x86/include/asm/vdso/gettimeofday.h
+++ b/arch/x86/include/asm/vdso/gettimeofday.h
@@ -271,24 +271,6 @@ static __always_inline const struct vdso_data *__arch_get_vdso_data(void)
 	return __vdso_data;
 }
 
-static inline bool arch_vdso_clocksource_ok(const struct vdso_data *vd)
-{
-	return true;
-}
-#define vdso_clocksource_ok arch_vdso_clocksource_ok
-
-/*
- * Clocksource read value validation to handle PV and HyperV clocksources
- * which can be invalidated asynchronously and indicate invalidation by
- * returning U64_MAX, which can be effectively tested by checking for a
- * negative value after casting it to s64.
- */
-static inline bool arch_vdso_cycles_ok(u64 cycles)
-{
-	return (s64)cycles >= 0;
-}
-#define vdso_cycles_ok arch_vdso_cycles_ok
-
 /*
  * x86 specific delta calculation.
  *
