fix for commit ab2be5794de8a57426ec3fc0f631cc766a0b8227

diff --git a/scripts/makepkg.sh.in b/scripts/makepkg.sh.in
index 8251e71b..cd1b2843 100644
--- a/scripts/makepkg.sh.in
+++ b/scripts/makepkg.sh.in
@@ -1071,8 +1090,8 @@ check_build_status() {
 				 && ! (( FORCE || SOURCEONLY || NOBUILD || NOARCHIVE)); then
 			if (( INSTALL )); then
 				warning "$(gettext "A package has already been built, installing existing package...")"
-				status=$(install_package)
-				exit $status
+				install_package
+				exit "$?"
 			else
 				error "$(gettext "A package has already been built. (use %s to overwrite)")" "-f"
 				exit $E_ALREADY_BUILT
@@ -1094,8 +1113,8 @@ check_build_status() {
 			if (( allpkgbuilt )); then
 				if (( INSTALL )); then
 					warning "$(gettext "The package group has already been built, installing existing packages...")"
-					status=$(install_package)
-					exit $status
+					install_package
+					exit "$?"
 				else
 					error "$(gettext "The package group has already been built. (use %s to overwrite)")" "-f"
 					exit $E_ALREADY_BUILT
